src/main/scala/GenerateExec.scala

```diff
@@ -0,0 +1,60 @@
+package fuselang
+
+import sys.process._
+import scala.io.Source
+import java.nio.file.{Files, Paths, Path, StandardOpenOption}
+
+import Errors.HeaderMissing
+
+/**
+ * Provides utilities to compile a program and link it with headers required
+ * by the CppRunnable backend.
+ */
+object GenerateExec {
+  val headers = List("parser.cpp", "picojson.h")
+
+  var headerLocation = Paths.get("src/main/resources/headers")
+  val headerFallbackLocation = Paths.get("_headers/")
```

Comment:
If the command is run from outside the compiler directory, `fuse run` will extract the required headers into a `_headers/` folder as a fallback.